File: python/retrieve_air_quality_data.py
import requests

airqloud_id = 'your_airqloud_id'  # Replace with the AirQloud ID for City X
access_token = 'your_access_token'  # Replace with your access token

api_url = f"https://api.airqo.net/api/v2/devices/measurements/airqlouds/{airqloud_id}?token={access_token}"

response = requests.get(api_url)

if response.status_code == 200:
    data = response.json()
    if 'data' in data:
        for measurement in data['data']:
            print(f"Date: {measurement['date']}")
            print(f"PM2.5: {measurement['pm25']}")
            # Include more data fields as needed
            print()
    else:
        print("Error: Unable to fetch data")
else:
    print("Error: Request failed")

File: php/retrieve-air-quality-data.php
<?php
$gridId = 'your_grid_id'; // Replace with the grid ID for City X
$accessToken = 'your_access_token'; // Replace with your access token

$apiUrl = "https://api.airqo.net/api/v2/devices/measurements/grids/$gridId?token=$accessToken";

$ch = curl_init($apiUrl);
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);

$response = curl_exec($ch);

if ($response) {
    $data = json_decode($response, true);
    if ($data && isset($data['data'])) {
        foreach ($data['data'] as $measurement) {
            echo "Date: " . $measurement['date'] . "\n";
            echo "PM2.5: " . $measurement['pm25'] . "\n";
            // Include more data fields as needed
            echo "\n";
        }
    } else {
        echo "Error: Unable to fetch data\n";
    }
} else {
    echo "Error: cURL request failed\n";
}

curl_close($ch);
?>
